shared/src/main/scala/amf/client/remod/AMFParser.scala

```diff
@@ -16,33 +16,31 @@ import amf.internal.resource.{ResourceLoader, StringResourceLoader}
 import scala.concurrent.Future
 import scala.scalajs.js.annotation.JSExport
 
-private[remod] object AMFParser {
-
+private[amf] object AMFParser {
 
   /**
     * Asynchronously generate a BaseUnit from the unit located in the given url.
     * @param url : Location of the api.
     * @param env: AnfEnvironment
     * @return A future that will have a BaseUnit or an error to handle the result of such invocation.
     */
-  def parse(url: String, env:BaseEnvironment): Future[AmfResult] = parseAsync(url, None, env)
+  def parse(url: String, env: AMFConfiguration): Future[AmfResult] = parseAsync(url, None, env)
 
   /**
     * Asynchronously generate a BaseUnit from the unit located in the given url.
     * @param url : Location of the api.
     * @return A future that will have a BaseUnit or an error to handle the result of such invocation.
     */
-
   // check Vendor , only param? ParseParams?
-  def parse(url: String, vendor:Vendor, env:BaseEnvironment): Future[AmfResult] = parseAsync(url, Some(vendor), env)
+  def parse(url: String, vendor: Vendor, env: AMFConfiguration): Future[AmfResult] = parseAsync(url, Some(vendor), env)
 
   /**
     * Asynchronously generate a BaseUnit from a given string.
     * @param stream: The unit as a string
     * @return A future that will have a BaseUnit or an error to handle the result of such invocation.
     */
   @JSExport
-  def parseStream(stream: String, env:AMFEnvironment): Future[AmfResult] = ???
+  def parseStream(stream: String, env: AMFConfiguration): Future[AmfResult] = ???
```

Comment:
Mmmh doubting about this. To me a `parseStream` method respects the premise that the result is also a Stream. What if I give AMF a never-ending Stream? Consuming the stream should be part of the client responsibility.

Reply:
Are you saying that a parseStream method should return a stream? that not makes so much sense in our context..
If the stream never ends, then is client fault. The same premise applies to a resource loader, a cache, o even a plugged parser.  Transformation pipelines could throw exceptions and that is another problem. If the user wants to break something, then I think there is no su much we could do, except for ban the possibility to inject logic in the framework but that is just what we want to have.

The only weird thing that I notice is that we are receiving and String not a real Stream, so all this discussion doesn't have sense. Stream logic could be handled using resource loaders and the method should be call "parseInMemory" "parseChain" "parseText" or something like that.
But as is a different discussion maybe the best approach is to ignored in here and create something focused on that.